code/game/objects/items/devices/radio/beacon.dm

```diff
@@ -64,21 +64,22 @@
 	hide(hides_under_flooring() && !T.is_plating())
 
 /obj/item/radio/beacon/anchored/attackby(obj/item/I, mob/living/user)
-	..()
 	if(istype(I, /obj/item/stack/nanopaste))
 		var/obj/item/stack/nanopaste/S = I
-		if(b_stat)
-			if(S.use(1))
-				to_chat(user, SPAN_NOTICE("You pour some of \the [S] over \the [src]'s circuitry."))
-				if(prob(repair_fail_chance))
-					flick("[initial(icon_state)]", src)
-					visible_message(SPAN_WARNING("The [src]'s lights come back on briefly, then die out again."), range = 2)
-				else
-					visible_message(SPAN_NOTICE("\The [src]'s lights come back on."), range = 2)
-					functioning = TRUE
-					repair_fail_chance += pick(5, 10, 10, 15)
-					update_icon()
-			else
-				to_chat(user, SPAN_WARNING("There's not enough of \the [S] left to fix \the [src]."))
+		if(!panel_open)
+			to_chat(user, SPAN_WARNING("You can't work on \the [src] until its been opened up."))
+			return TRUE
+		if(!S.use(1))
+			to_chat(user, SPAN_WARNING("There's not enough of \the [S] left to fix \the [src]."))
+			return TRUE
+		to_chat(user, SPAN_NOTICE("You pour some of \the [S] over \the [src]'s circuitry."))
+		if(prob(repair_fail_chance))
+			flick("[initial(icon_state)]", src)
+			visible_message(SPAN_WARNING("The [src]'s lights come back on briefly, then die out again."), range = 2)
 		else
-			to_chat(user, SPAN_WARNING("You can't work on \the [src] until its been opened up."))
\ No newline at end of file
+			visible_message(SPAN_NOTICE("\The [src]'s lights come back on."), range = 2)
+			functioning = TRUE
+			repair_fail_chance += pick(5, 10, 10, 15)
```

Comment:
imo if this goes over 100, it should says so in examine or at beginning of attempt, so people aren't stuck trying to repair a busted-forever radio.

Reply:
Added a check to attackby